librz/util/str.c

```diff
@@ -753,22 +753,25 @@ RZ_API char *rz_str_newf(const char *fmt, ...) {
 	return p;
 }
 
-// Secure string copy with null terminator (like strlcpy or strscpy but ours
+/**
+ * \brief Secure string copy with null terminator
+ *
+ * 	This API behaves like strlcpy or strscpy.
+ */
 RZ_API size_t rz_str_ncpy(char *dst, const char *src, size_t n) {
 	rz_return_val_if_fail(dst && src, 0);
-	size_t i;
 
 	// do not do anything if n is 0
 	if (n == 0) {
 		return 0;
 	}
-
-	n--;
-	for (i = 0; src[i] && n > 0; i++, n--) {
-		dst[i] = src[i];
-	}
-	dst[i] = 0;
-	return i;
+#if HAVE_STRLCPY
+	return strlcpy(dst, src, n);
+#else
+	strncpy(dst, src, n - 1);
+	dst[n - 1] = '\0';
+	return n;
```

Comment:
This is not the correct return value. It should be the same as the one of strlcpy. Check the man page of strlcpy to understand what to return